client/api_bucket.go

```diff
@@ -378,10 +378,34 @@ func (c *client) ListBuckets(ctx context.Context) (types.ListBucketsResult, erro
 		disableCloseBody: true,
 	}
 
-	endpoint, err := c.getInServiceSP()
-	if err != nil {
-		log.Error().Msg(fmt.Sprintf("get in-service SP fail %s", err.Error()))
-		return types.ListBucketsResult{}, err
+	var endpoint *url.URL
+	var useHttps bool
+	var err error
+	if opts.Endpoint != "" {
+		if strings.Contains(opts.Endpoint, "https") {
```

Comment:
please turn the code into a function. for example, getEndpoint (opts listOption),